packages/common/src/store/pages/chat/slice.ts

```diff
@@ -73,7 +73,30 @@ export const chatMessagesAdapter = createEntityAdapter<ChatMessageWithExtras>({
   sortComparer: messageSortComparator
 })
 
-const { selectById: getMessage } = chatMessagesAdapter.getSelectors()
+const {
+  selectById: getMessage,
+  selectEntities: getAllMessages,
+  selectIds: getAllMessageIds
+} = chatMessagesAdapter.getSelectors()
+
+// Recalculate hasTail for the message at index - 1 (the previous message of
```

Comment:
nit: maybe update comment and say index + 1 and explain that message oder is backwards?